optuna/notebooks/jupytercon_demo/RAPIDS_xfeat_Optuna-GPU.ipynb

```diff
@@ -0,0 +1,1803 @@
+    "    # Xfeat's internal fold mechanism creates RangeIndex references, so we need to do an index reset on our data frames.\n",
+    "    df_train = df_train.reset_index(drop=True)\n",
+    "    df_test = df_test.reset_index(drop=True)\n",
+    "    y_train = y_train.reset_index(drop=True)\n",
+    "    y_test = y_test.reset_index(drop=True)\n",
+    "    \n",
+    "    # Need to do this to ensure we are appropriately assigning the split values\n",
+    "    df_train[TARGET_COL] = y_train\n",
+    "    df_test[TARGET_COL] = y_test\n",
+    "    \n",
+    "    for col in CAT_COLS:\n",
+    "        out_col = f'{col}_TE'\n",
+    "        lbl_enc = LabelEncoder(handle_unknown='ignore')\n",
+    "        tar_enc = TargetEncoder(n_folds=5, smooth=TARGET_ENC_SMOOTH, split_method=TARGET_ENC_SPLIT)\n",
+    "\n",
+    "        df_train[col] = lbl_enc.fit_transform(df_train[col])\n",
+    "        df_train[out_col] = tar_enc.fit_transform(df_train[col], df_train[TARGET_COL])\n",
+    "\n",
+    "        df_test[out_col] = tar_enc.transform(df_test[col])\n",
+    "        df_test[col] = lbl_enc.transform(df_test[col]).fillna(0)\n",
+    "        del lbl_enc, tar_enc\n",
+    "\n",
+    "    encoder = Pipeline([\n",
+    "                        SelectNumerical(),\n",
+    "                        ArithmeticCombinations(exclude_cols=[TARGET_COL],\n",
+    "                                               drop_origin=False,\n",
+    "                                               operator=\"+\",\n",
+    "                                               r=2,\n",
+    "                                               output_suffix=\"_plus\")\n",
+    "                    ])\n",
+    "\n",
+    "    df_train = encoder.fit_transform(df_train)\n",
+    "    df_test = encoder.transform(df_test)\n",
+    "    df = cudf.concat([df_train, df_test], sort=False)\n",
+    "    return df"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Feature Selection and Hyper parameter Optimisation\n",
+    "\n",
+    "Now that we have some new features, how do we know they are relevant for the task or represent anything meaningful? We use the feature selection process to do this. This helps in selection of a subset of features that are most informative. This helps in simplifying the problem and ensures that we aren't overloading the system with unimportant features.\n",
+    "\n",
+    "Optuna provides a way to choose a selector which accepts a Pipeline object from xfeat. You can see in the `feature_selection` function we define a `Pipeline` that takes in an Explorer and a Selection Algorithm (`ChiSquareKBest`). We pass this to an Optuna Study object, along with an Objective function. Chi squared tests are used to test the independence of two events.\n",
+    "\n",
+    "For Feature Selection, we aim to select features, which are highly dependent on the response. This way, we can get features that will best determine the outcome.\n",
```

Comment:
```suggestion
    "For Feature Selection, we aim to pick only those features that are most useful in predicting the response.\n",
```

Reply:
This is actually the explanation for Chi-square test. That's how it performs feature selection, I'll word it better to make this clear.